### ISSUE TITLE
Problem with interval report in Analyze plugin 

### ISSUE THREAD

**Issue by leobin** [tsduck/tsduck#48]
When using interval report options in Analyze plugin. It doesn't output at the right interval.
I took a look at plugin code and I think the problem is in this part of code
```
 else if (_next_report_packet == 0 || (_next_report_packet > 0 && _current_packet >= _next_report_packet)) {
            // Check current time to see if this is time to produce a report
            const Time current_utc(Time::CurrentUTC());
            if (current_utc < _next_report_time) {
                // False alarm, we have to wait some more
                computeNextReportTime(current_utc, _next_report_time - current_utc);
            }
            else {
                // Time to produce a report
                if (!produceReport()) {
                    return TSP_END;
                }
                // Reset analysis context
                _analyzer.reset();
                computeNextReportTime (current_utc, _output_interval);
            }
        }
```

In my case, my stream doesn't have bit-rate information. 
In the above code, it's always a wrong report anytime we receive a packet. 
So It will run computeNextReportTime(current_utc, _next_report_time - current_utc); every time we receive a packet. 
And in case we keep calling function computeNextReportTime, which I think reduce next_report_time a little bit for every call. We end up with wrong next_report time
So I think it should be like this:
  
```
if (current_utc < _next_report_time) {
    // False alarm, we have to wait some more
    if (_next_report_packet > 0) { //don't recompute when don't have bitrate information.
        computeNextReportTime(current_utc, _next_report_time - current_utc);
    }
}
```

**Comment by lelegard** [comment]
Hi,

Thank you for reporting this. It is a mistake and I will fix it.

In fact, the mistake is slightly different. When processing a transport stream, there are two fundamentally different types of time intervals or durations : based on the wall clock time and based on the transport stream clock. When working on a live stream, the two are identical. But when working on a TS file, they are completely different, the wall clock time is based on the file reading speed.

For instance, when we need to insert a section with a repetition rate of 10 ms, it is obviously based on the TS time. But, for the plugin `regulate`, the time is the wall clock time. So, depending on the use case, we have to select the right reference.

With `-P analyze --interval`, we have to define what we mean by time. But the most sensible use case is that there is a very long  processing (possibly infinite on a live stream) and we want get a snapshot of what is going on in the TS from time to time. So, this is wall clock time.

And obviously, I made a mistake because the two are mixed: computing the next insertion packet based on the TS bitrate, which is the standard technique when playing with the TS time, and checking the UTC wall clock time. I must have been completely out of my mind when I wrote this. I have always used the `--interval` option on live streams, real TV streams with PCR's, so I never noticed the difference.

Additionally, your remark on the non-regularity of the interval is correct. Honestly, the use case here is to get a snapshot from time to time, without real-time constraint. But you are right in principle and it deserves a change. I will use an instance of `ts::Monotonic` as done in plugin `regulate` which uses a much more precise synchronisation.

I will update this thread when I fix the issue.
